src/CacheManager.h

```diff
@@ -49,6 +50,11 @@ class CacheManager
     static CacheManager* GetInstance();
     const char *ActionProtection(const Mgr::ActionProfilePointer &profile);
 
+    /// Add HTTP response headers specific/common to all cache manager replies,
+    /// including cache manager errors and Action reports.
+    /// \param httpOrigin the value of Origin header in the trigger HTTP request (or nil)
+    static void PutCommonResponseHeaders(HttpReply &, const char *httpOrigin);
+
```

Comment:
> IMO, the request-time changes in the PR are valuable on their own and should stay because
> 
>     * The sooner Squid knows that the response is going to be uncachable, the better.

mgr responses being non-cacheable is not guaranteed. As I mentioned earlier there are multiple cases where the response SHOULD be cached. All these request-time changes make it much harder to correct those responses caching in future.

> This knowledge helps reduce cache manager request impact on Store state, which may be very valuable in triage and tests. 

I disagree on the utility. It is certain to add one more case to the existing difficulty troubleshooting why apparently cacheable mgr responses from upstream servers are not being stored. Whereas it is easy to tell from upstream headers when a response is incorrectly made cacheable by them.

> Most clients sending cache manager requests do not tell Squid that the responses are not cachable,

Irrelevant. Cacheability in HTTP is determined by the server. At most a client can request a cache be bypassed and/or refreshed.

>     * Those request-time changes fix a known problem with flags.cachable getting overwritten by maybeCacheable(), reduce code duplication of tricky code, improve debugging, improve code documentation, and make Optional trivially copyable. They are a step forward on many levels, and they are not complex.
> 

Maybe, BUT, none of that extra change is actually fixing the cache manager response bug. It would make review and change documentation a lot better if each of those issue fixes were isolated in their own PRs, or at least PRs scoped more tightly to each particular problematic object.

> > Or maybe even just document a suggestion to admin for store_miss ACL rules to workaround it.
> 
> I agree that warning admins that use affected cache manager requests and sharing this workaround is a good idea.
> Once this PR lands, I can volunteer to post a message to squid-users about it if you think that is the best way to
> document that suggestion.

A post to squid-announce is probably better. That would go to squid-users plus a bunch of others.

For backport I would like the change of generated headers, but the rest is unclear at present.

Reply:
> Alex: Once this PR lands, I can volunteer to post a message to squid-users about it if you think that is the best way to document that suggestion.

> Amos: A post to squid-announce is probably better. That would go to squid-users plus a bunch of others.

Sounds good, will do.